File: 0414-third-maximum-number/solution.cpp
class Solution {
public:
    int thirdMax(vector<int>& nums) {
        int n = nums.size();
        set<int> st;
        for(int i = 0; i < n; i++){
            st.insert(nums[i]);
        }
        vector<int> arr(st.begin(), st.end());
        int m = arr.size();
            if(m <= 2){
                return arr[m-1];
            }
        return arr[m-3];
    }
};

File: 0414-third-maximum-number/solution.java
class Solution {
    public int thirdMax(int[] nums) {
        Set<Integer> max = new TreeSet<>();
        for(int i = 0; i < nums.length; i++){
            max.add(nums[i]);
        }
        ArrayList<Integer> arr = new ArrayList<>(max);
        if(arr.size() <= 2) return arr.get(arr.size() - 1);
        if(arr.size() == 3) return arr.get(0);
        return arr.get(arr.size()-3);
    }
}
